core/rts.cpp

```diff
@@ -48,6 +52,17 @@ void RelationalTransitionSystem::constrain_trans(const Term & constraint)
     throw PonoException("Unknown symbols");
   }
   trans_ = solver_->make_term(And, trans_, constraint);
+  set_updated_states(constraint);
 }
 
+void RelationalTransitionSystem::set_updated_states(const smt::Term & term)
+{
+  UnorderedTermSet free_vars;
+  get_free_symbolic_consts(term, free_vars);
+  for (const auto & free_var : free_vars) {
+    if (next_statevars_.count(free_var)) {
+      no_state_updates_.erase(free_var);
+    }
```

Comment:
There seems to be a mistake. The next-state var has to mapped back to curr-state var before erasing.
Could you double-check?

```suggestion
    if (next_statevars_.find(free_var) != next_state_vars.end()) {
      no_state_updates_.erase(curr_map[free_var]);
    }
```

Reply:
You are correct, I was missing getting from the curr_map. Thanks for the thorough review.
Fixed at a9fa4b597cc5fdc2fe14f44a4fcbd84308e7c8ef.
Also adding the unit test and some of the original tests from #475 as they are required for covering this fix.